### PR TITLE
Support for optional replacing of record's timestamp

### PR DESCRIPTION
**Pull request by bohyn**
Support for optional replacing of record's timestamp by value of 'time' field.

### CODE REVIEW

File: lib/fluent/plugin/in_kafka.rb [hunk 1]
```diff
@@ -31,6 +31,10 @@ class Fluent::KafkaInput < Fluent::Input
 
   config_param :offset_zookeeper, :string, :default => nil
   config_param :offset_zk_root_node, :string, :default => '/fluent-plugin-kafka'
+  config_param :input_replace_time, :bool, :default => false,
```
Comment (repeatedly) [inline lib/fluent/plugin/in_kafka.rb]: `use_record_time` is more clear

File: lib/fluent/plugin/in_kafka.rb [hunk 2]
```diff
@@ -231,7 +235,16 @@ def consume
 
       messages.each { |msg|
         begin
-          es.add(Fluent::Engine.now, @parser.call(msg, @topic_entry))
+          record = @parser.call(msg, @topic_entry)
+          record_time = Fluent::Engine.now
+          if @input_replace_time
+            if @time_format
+              record_time = Time.strptime(record['time'], @time_format)
```
Comment (repeatedly) [inline lib/fluent/plugin/in_kafka.rb]: Don't use `Time.strptime`.
Use Fluentd's `TimeParser` instead.

File: lib/fluent/plugin/in_kafka.rb [hunk 3]
```diff
@@ -231,7 +235,16 @@ def consume
 
       messages.each { |msg|
         begin
-          es.add(Fluent::Engine.now, @parser.call(msg, @topic_entry))
+          record = @parser.call(msg, @topic_entry)
+          record_time = Fluent::Engine.now
```
Comment (repeatedly) [inline lib/fluent/plugin/in_kafka.rb]: Don't create `Fluent::Engine.now` when option is true.
It allocates unnecessary object.

File: lib/fluent/plugin/in_kafka.rb [hunk 4]
```diff
@@ -231,7 +236,18 @@ def consume
 
       messages.each { |msg|
         begin
-          es.add(Fluent::Engine.now, @parser.call(msg, @topic_entry))
+          record = @parser.call(msg, @topic_entry)
+          if @use_record_time
+            if @time_format
+              time_parser = Fluent::TextParser::TimeParser.new(@time_format)
```
Comment (repeatedly) [inline lib/fluent/plugin/in_kafka.rb]: Creating TimeParser in `each` is very high cost.
It should be created in `configure` when `use_record_time` is `true`

Comment (bohyn) [inline lib/fluent/plugin/in_kafka.rb]: Fixed

**Comment by repeatedly** [comment]
Thanks!

**Pull request merged**